feat(dmarc-reporter): add dmarc reporter

andreialecu opened the pull request:
This is just a draft for now.

Closes #230

jogold commented on src/dmarc/index.ts:
```diff
@@ -0,0 +1,112 @@
+   * The email address to send DMARC reports to.
+   * This email address must be verified in SES.
+   */
+  emailAddress: string;
```
Can we default to `dmarc-reports` @ the domain name of the provided hosted zone?

jogold commented on src/dmarc/index.ts:
```diff
@@ -0,0 +1,112 @@
+   * - quarantine: Quarantine messages that fail DMARC compliance.
+   * - reject: Reject messages that fail DMARC compliance.
+   */
+  dmarcPolicy: 'none' | 'quarantine' | 'reject';
```
Let's create an enum.

```ts
export enum DmarcPolicy {
  NONE = 'none',
  ...
}
```

can we have default value?

jogold commented on src/dmarc/index.ts:
```diff
@@ -0,0 +1,112 @@
+   * - reject: Reject messages that fail DMARC compliance.
+   * @default inherited from dmarcPolicy
+   */
+  dmarcSubdomainPolicy?: 'none' | 'quarantine' | 'reject';
```
enum

jogold commented on src/dmarc/index.ts:
```diff
@@ -0,0 +1,112 @@
+   * - relaxed: Use relaxed alignment mode.
+   * - strict: Use strict alignment mode.
+   */
+  dmarcDkimAlignment?: 'relaxed' | 'strict';
```
enum

jogold commented on src/dmarc/index.ts:
```diff
@@ -0,0 +1,112 @@
+   * - relaxed: Use relaxed alignment mode.
+   * - strict: Use strict alignment mode.
+   */
+  dmarcSpfAlignment?: 'relaxed' | 'strict';
```
enum

jogold commented on src/dmarc/index.ts:
```diff
@@ -0,0 +1,112 @@
+   * The properties to pass to the EmailReceiver construct.
+   * The `recipients` property will be ignored.
+   */
+  emailReceiverProps: Omit<EmailReceiverProps, 'recipients'>;
```
Unfortunately the `Omit` will not work with JSII (used to provide the constructs in all the languages supported by the CDK), you will have to expose `receiptRuleSet`, `function` and `afterRule` in this interface.

jogold reviewed (commented):
Nice! See my comments and thank you already for the contribution!

andreialecu commented:
> Nice! See my comments and thank you already for the contribution!

Updated. If you'd like to merge it like this I could set it as Ready for review.

I was thinking about doing a full implementation using https://github.com/andreialecu/dmarc-report-parser but that could be left to the user I guess.

andreialecu commented:
Added a snapshot test and updated docs as well. Should be good to go now. Up to you how you want to move forward with it. Let me know if you need anything else!

I'll probably just add something to the readme in https://github.com/andreialecu/dmarc-report-parser about how it can tie into this construct (and maybe update the docs for it here to link back there). But otherwise, I guess this is good to go like this.

jogold reviewed (approved):
Great!
